src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/HtmlController.kt

```diff
@@ -31,11 +28,30 @@ class HtmlController(@Autowired val dashboardService: DashboardService) {
         return RedirectView("/dashboard")
     }
 
+    @RequestMapping("/choose-supplier")
+    fun chooseSupplier(model: ModelMap): String {
+        return "choose-supplier"
+    }
+
+    @RequestMapping("/choose-supplier/serco")
+    fun chooseSupplierSerco(session: HttpSession): RedirectView {
+        session.setAttribute("supplier", Supplier.SERCO)
+        return RedirectView("/dashboard")
+    }
+
+    @RequestMapping("/choose-supplier/geoamey")
+    fun chooseSupplierGeoAmey(session: HttpSession): RedirectView {
+        session.setAttribute("supplier", Supplier.GEOAMEY)
+        return RedirectView("/dashboard")
+    }
+
     @RequestMapping("/dashboard")
-    fun dashboard(@RequestParam(name = "date", required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) date: LocalDate?, model: ModelMap): String {
+    fun dashboard(@SessionAttribute(name = "date") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) date: LocalDate?, @SessionAttribute(name = "supplier") supplier: Supplier?, model: ModelMap): Any {
+        if (supplier == null) {
```

Comment:
Because this is going to be a repeated pattern per RequestMapping (apart from choose-supplier), probably better to put the check for supplier in a Filter - see https://stackoverflow.com/questions/57137095/spring-mvc-session-check-before-every-request